src/channel/NodeChannel.js

```diff
@@ -13,15 +14,36 @@ export default class NodeChannel extends Channel {
     /**
      * @internal
      * @param {string} address
+     * @param {string | undefined} certificate
      */
-    constructor(address) {
+    constructor(address, certificate) {
         super();
 
+        let clientCredentials;
+        let clientAddress;
+
+        if (certificate != null) {
+            clientCredentials = credentials.createSsl(
+                Buffer.from(utf8.encode(certificate))
+            );
+            clientAddress = `${address.split(":")[0]}:50212`;
+        } else if (!address.endsWith(":50212")) {
+            clientCredentials = credentials.createInsecure();
+            clientAddress = `${address.split(":")[0]}:50211`;
```

Comment:
Same as before for non-TLS version. Specifically when I run the tests against Hedera Services master the network is `127.0.0.1:50213..4` , this would make those tests fail.